pytket/pytket/qasm/qasm.py

```diff
@@ -1142,6 +1142,12 @@ def check_can_convert_circuit(circ: Circuit, header: str, maxwidth: int) -> None
             f"Circuit contains a classical register larger than {maxwidth}: try "
             "setting the `maxwidth` parameter to a higher value."
         )
+    set_circ_register = set([creg.name for creg in circ.c_registers])
+    for b in set([b.reg_name for b in circ.bits]):
+        if b not in set_circ_register:
+            raise QASMUnsupportedError(
+                f"Circuit contains a invalid classical register {b}."
```

Comment:
```suggestion
    for b in circ.bits:
        if b.reg_name not in set_circ_register:
            raise QASMUnsupportedError(
                f"Circuit contains a invalid classical register {b.reg_name}."
```

Reply:
done in [56ad98d](https://github.com/CQCL/tket/pull/1656/commits/56ad98d0ee5fbe6550b2ea24d681bd93170ba342)